src/client/datascience/interactive-ipynb/nativeEditor.ts

```diff
@@ -439,13 +441,29 @@ export class NativeEditor extends InteractiveBase implements INotebookEditor {
      * @memberof NativeEditor
      */
     private async updateVersionInfoInNotebook(): Promise<void> {
-        // Make sure we have notebook json
-        await this.ensureNotebookJson();
+        // Get our kernel_info and language_info from the current notebook
+        const notebook = this.getNotebook();
 
-        // Use the active interpreter
-        const usableInterpreter = await this.jupyterExecution.getUsableJupyterPython();
-        if (usableInterpreter && usableInterpreter.version && this.notebookJson.metadata && this.notebookJson.metadata.language_info) {
-            this.notebookJson.metadata.language_info.version = `${usableInterpreter.version.major}.${usableInterpreter.version.minor}.${usableInterpreter.version.patch}`;
+        if (notebook) {
+            const interpreter = await notebook.getMatchingInterpreter();
+            const kernelSpec = await notebook.getKernelSpec();
```

Comment:
No need to make them sequential.
```suggestion
            const [interpreter, kernelSpec]  = await Promise.all([notebook.getMatchingInterpreter(), notebook.getKernelSpec()]);
```